src/Razor/src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpLanguageCharacteristics.cs

```diff
@@ -64,8 +62,105 @@ internal class CSharpLanguageCharacteristics : LanguageCharacteristics<CSharpTok
+            { CSharpKeyword.Ushort, "ushort" },
+            { CSharpKeyword.While, "while" },
+            { CSharpKeyword.Case, "case" },
+            { CSharpKeyword.Const, "const" },
+            { CSharpKeyword.Explicit, "explicit" },
+            { CSharpKeyword.Float, "float" },
+            { CSharpKeyword.Null, "null" },
+            { CSharpKeyword.Sizeof, "sizeof" },
+            { CSharpKeyword.Typeof, "typeof" },
+            { CSharpKeyword.Implicit, "implicit" },
+            { CSharpKeyword.Private, "private" },
+            { CSharpKeyword.This, "this" },
+            { CSharpKeyword.Using, "using" },
+            { CSharpKeyword.Extern, "extern" },
+            { CSharpKeyword.Return, "return" },
+            { CSharpKeyword.Stackalloc, "stackalloc" },
+            { CSharpKeyword.Uint, "uint" },
+            { CSharpKeyword.Base, "base" },
+            { CSharpKeyword.Catch, "catch" },
+            { CSharpKeyword.Continue, "continue" },
+            { CSharpKeyword.Double, "double" },
+            { CSharpKeyword.For, "for" },
+            { CSharpKeyword.In, "in" },
+            { CSharpKeyword.Lock, "lock" },
+            { CSharpKeyword.Object, "object" },
+            { CSharpKeyword.Protected, "protected" },
+            { CSharpKeyword.Static, "static" },
+            { CSharpKeyword.False, "false" },
+            { CSharpKeyword.Public, "public" },
+            { CSharpKeyword.Sbyte, "sbyte" },
+            { CSharpKeyword.Throw, "throw" },
+            { CSharpKeyword.Virtual, "virtual" },
+            { CSharpKeyword.Decimal, "decimal" },
+            { CSharpKeyword.Else, "else" },
+            { CSharpKeyword.Operator, "operator" },
+            { CSharpKeyword.String, "string" },
+            { CSharpKeyword.Ulong, "ulong" },
+            { CSharpKeyword.Bool, "bool" },
+            { CSharpKeyword.Char, "char" },
+            { CSharpKeyword.Default, "default" },
+            { CSharpKeyword.Foreach, "foreach" },
+            { CSharpKeyword.Long, "long" },
+            { CSharpKeyword.Void, "void" },
+            { CSharpKeyword.Enum, "enum" },
+            { CSharpKeyword.Finally, "finally" },
+            { CSharpKeyword.Int, "int" },
+            { CSharpKeyword.Out, "out" },
+            { CSharpKeyword.Sealed, "sealed" },
+            { CSharpKeyword.True, "true" },
+            { CSharpKeyword.Goto, "goto" },
+            { CSharpKeyword.Unchecked, "unchecked" },
+            { CSharpKeyword.Interface, "interface" },
+            { CSharpKeyword.Break, "break" },
+            { CSharpKeyword.Checked, "checked" },
+            { CSharpKeyword.Namespace, "namespace" },
+            { CSharpKeyword.When,  "when" },
+        };
+
+        private static readonly CSharpLanguageCharacteristics _instance = new CSharpLanguageCharacteristics();
+
         protected CSharpLanguageCharacteristics()
         {
+#if DEBUG
+            for (int i = (int) CSharpKeyword.Await; i <= (int) CSharpKeyword.When; i++)

```

Comment:
This means if CSharpKeyword increases in size, this will not fail right?

Reply:
Yeah, I was being lazy. Updated the verification code to not have understanding of the enumeration names.